### PR TITLE
PullRequest for Issue8: add feature for StartTime to EndTime request type

### PR DESCRIPTION
**Pull request by XuetaoLiu-S**
Add the support of StartTimeToEndTime dataRequest type and refactor the codes to make the codes more clear.
https://github.com/acquaman/AcquamanDataServer/issues/8

### CODE REVIEW

**Comment by iainworkman** [comment]
Also looks good.
Time Estimate: 30 minutes

File: source/AMDSBufferGroup.cpp [hunk 1]
```diff
@@ -83,15 +94,23 @@ void AMDSBufferGroup::populateData(AMDSClientRelativeCountPlusCountDataRequest*
...
+void AMDSBufferGroup::populateData(AMDSClientStartTimeToEndTimeDataRequest* clientDataRequest)
+{
+	QDateTime startTime = clientDataRequest->startTime();
+	QDateTime endTime = clientDataRequest->endTime();
 
-	for (int iCurrent = startIndex; iCurrent < endIndex; iCurrent++)
-	{
-		AMDSDataHolder* dataHolder = dataHolders_[iCurrent];
-		clientDataRequest->appendData(dataHolder);
+	int startIndex = lowerBound(startTime);
```
Comment (dretrex) [inline source/AMDSBufferGroup.cpp:114]: I had a quick look at it, but how expensive is calling `lowerBound()`?  It looks like it's almost like a quick sort.

File: source/ClientRequest/AMDSClientDataRequest.cpp [hunk 2]
```diff
@@ -187,3 +187,15 @@ bool AMDSClientDataRequest::readFromDataStream(AMDSDataStream *dataStream)
 
 	return true;
 }
+
+void AMDSClientDataRequest::printData()
+{
+	QList<AMDSFlatArray> values;
+	for(int x = 0, size = data().count(); x < size; x++){
+		AMDSFlatArray oneFlatArray = AMDSFlatArray(uniformDataType(), bufferGroupInfo().spanSize());
+//		AMDSFlatArray oneFlatArray;
```
Comment (dretrex) [inline source/ClientRequest/AMDSClientDataRequest.cpp]: Comment that can be removed.

**Comment by dretrex** [comment]
Overall, looks good.
Time estimate: 15 minutes

**Pull request merged**